src/core/tests/type_prop/assign.cpp

```diff
@@ -78,3 +79,99 @@ TEST(type_prop, variable_comparison) {
     ASSERT_FALSE(variable1->get_info() == variable4->get_info());
     ASSERT_FALSE(variable1->get_info() == variable5->get_info());
 }
+
+TEST(type_prop, assign_v6_static_shape_match) {
+    auto input = make_shared<ov::op::v0::Parameter>(element::f32, Shape{1, 2, 64, 64});
+    auto variable = std::make_shared<op::util::Variable>(
+        op::util::VariableInfo{PartialShape{1, 2, 64, 64}, element::f32, "variable_id"});
+    std::shared_ptr<ov::op::v6::Assign> assign;
+    EXPECT_NO_THROW(assign = std::make_shared<ov::op::v6::Assign>(input, variable));
+
+    ASSERT_EQ(assign->get_element_type(), element::f32);
+    ASSERT_EQ(assign->get_shape(), (Shape{1, 2, 64, 64}));
+}
+
+TEST(type_prop, assign_v6_static_shapes_do_not_match) {
```

Comment:
Add test to check if dimensions labels are propagated or not from inputs to outputs?